statement.go

```diff
@@ -36,15 +73,56 @@ func (s *stmt) Close() error {
 	return nil
 }
 
-func (s *stmt) NumInput() int {
+// NumInput returns the number of placeholders in the statement.
+// It implements the driver.Stmt interface.
+func (s *Stmt) NumInput() int {
 	if s.closed {
 		panic("database/sql/driver: misuse of duckdb driver: NumInput after Close")
 	}
 	paramCount := C.duckdb_nparams(*s.stmt)
 	return int(paramCount)
 }
 
-func (s *stmt) bind(args []driver.NamedValue) error {
+// ParamName returns the name of the parameter at the given index (1-based).
+func (s *Stmt) ParamName(n int) string {
+	if s.closed {
```

Comment:
Let's add a test triggering this error. Does this correctly handle indexes that are out of bounds? I.e., should it return an empty string or an error? Same for `ParamType`.